src/Uno.UI/UI/Xaml/Style/Style.cs

```diff
@@ -67,6 +67,9 @@ internal void ApplyTo(DependencyObject o, DependencyPropertyValuePrecedences pre
 					{
 						pair.Value(o);
 					}
+
+					// Check tree for resource binding values, since some Setters may have set ThemeResource-backed values
+					(o as IDependencyObjectStoreProvider).Store.UpdateResourceBindings(isThemeChangedUpdate: false);
```

Comment:
I like the readability of the safe cast, but a null ref in logs is much less helpful than an invalid cast exception which dumps the source and target type ...

Reply:
In this case `DependencyObject` is guaranteed to be a `IDependencyObjectStoreProvider` by the backing generator, the safe cast is just to access the `Store` property (which is explicitly implemented)